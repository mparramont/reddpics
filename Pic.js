import React from "react";
import { Image, StyleSheet, Text, TouchableOpacity, View } from "react-native";
import Moment from "react-moment";
import "moment-timezone";

export default class Pic extends React.Component {
  onPressListItem(url, title) {
    this.props.navigation.navigate("Details", {
      url: url,
      title: title
    });
  }

  render() {
    let title = this.props.item.data.title;
    if (title.length > 50) title = title.substring(0, 50) + "â€¦";

    return (
      <TouchableOpacity
        style={{
          flexDirection: "row",
          paddingTop: 8,
          paddingLeft: 4,
          paddingRight: 4,
          paddingBottom: 8,
          borderColor: "tomato",
          borderBottomWidth: 1
        }}
        onPress={() =>
          this.onPressListItem(
            this.props.item.data.url,
            this.props.item.data.title
          )
        }
      >
        <Image
          style={{ width: "33%", height: 100 }}
          source={{ uri: this.props.item.data.thumbnail }}
        />
        <View
          style={{
            width: "67%",
            height: 100,
            padding: 5,
            justifyContent: "space-between"
          }}
        >
          <View style={{ marginLeft: "auto" }}>
            <Moment fromNow unix tz="Europe/Madrid" element={Text}>
              {this.props.item.data.created_utc}
            </Moment>
          </View>
          <View>
            <Text style={{ fontSize: 18 }}>{title}</Text>
          </View>
          <View
            style={{
              flexDirection: "row",
              justifyContent: "space-between"
            }}
          >
            <Text>{this.props.item.data.author}</Text>
            <Text>{this.props.item.data.score}</Text>
            <Text>{this.props.item.data.num_comments}</Text>
          </View>
        </View>
      </TouchableOpacity>
    );
  }
}
