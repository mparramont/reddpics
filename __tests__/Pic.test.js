import "react-native";
import React from "react";
import Pic from "../Pic";

// Note: test renderer must be required after react-native.
import renderer from "react-test-renderer";

let navigation = { navigate: jest.fn() };
let item = {
  data: {
    author: "omnicot",
    score: 19440,
    thumbnail:
      "https://b.thumbs.redditmedia.com/-LqcIIv3SFEuUvqKArGvIHxuasQPU6EApGZKmo6YLog.jpg",
    url: "https://i.redd.it/199nszs6w7e01.png",
    title: "some nice pic!",
    created_utc: 1517759326,
    num_comments: 422
  }
};

test("renders correctly", () => {
  const tree = renderer
    .create(<Pic navigation={navigation} item={item} />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});

test("with a pretty long title, truncates it", () => {
  item.data.title =
    "This title is more than 50 characters long, so it will get truncated";
  const tree = renderer
    .create(<Pic navigation={navigation} item={item} />)
    .toJSON();
  const treeString = JSON.stringify(tree);

  expect(treeString).toEqual(
    expect.stringContaining(
      "This title is more than 50 characters long, so it â€¦"
    )
  );
});
