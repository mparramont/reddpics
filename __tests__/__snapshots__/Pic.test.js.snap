// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "borderBottomWidth": 1,
      "borderColor": "tomato",
      "flexDirection": "row",
      "opacity": 1,
      "paddingBottom": 8,
      "paddingLeft": 4,
      "paddingRight": 4,
      "paddingTop": 8,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <Image
    source={
      Object {
        "uri": "https://b.thumbs.redditmedia.com/-LqcIIv3SFEuUvqKArGvIHxuasQPU6EApGZKmo6YLog.jpg",
      }
    }
    style={
      Object {
        "height": 100,
        "width": "33%",
      }
    }
  />
  <View
    style={
      Object {
        "height": 100,
        "justifyContent": "space-between",
        "padding": 5,
        "width": "67%",
      }
    }
  >
    <View
      style={
        Object {
          "marginLeft": "auto",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        dateTime={"2018-02-04T15:48:46.000Z"}
        ellipsizeMode="tail"
      >
        2 days ago
      </Text>
    </View>
    <View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 18,
          }
        }
      >
        some nice pic!
      </Text>
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        omnicot
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        19440
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        422
      </Text>
    </View>
  </View>
</View>
`;
